sdk/azcore/poller.go

```diff
@@ -0,0 +1,453 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	headerOperationLocation = "Operation-Location"
+	headerLocation          = "Location"
+)
+
+// ErrorUnmarshaller is the func to invoke when the endpoint returns an error response that requires unmarshalling.
+type ErrorUnmarshaller func(*Response) error
+
+// NewLROPoller creates an LROPoller based on the provided initial response.
+// NOTE: this is only meant for internal use in generated code.
+func NewLROPoller(pollerType string, resp *Response, pl Pipeline, eu ErrorUnmarshaller) (*LROPoller, error) {
```

Comment:
Should we document what pollerType should be??

Reply:
I added a doc comment.  I also renamed the param to `pollerID` as I felt that was a more accurate description.